external/CMakeLists.txt

```diff
@@ -0,0 +1,55 @@
+# -*- mode: cmake; -*-
+# - osquery external project builds
+#
+# Symlink directories for extensions or modules here.
+# If a directory includes: "extension_NAME" it is built as an extension.
+# If a directory includes: "module_NAME" it is built as a module.
+
+add_custom_target(externals)
+add_definitions(-DOSQUERY_EXTERNAL)
```

Comment:
This call should be replaced with `target_compile_definitions`

Reply:
I would like to add the compile option to the directory property instead of a specific target. Can I use the `target_compile_definitions` for it? I changed it to `add_compile_options`.

Reply:
The `add_compile_options` sets the options not only on the current folder but also on everything else below it. Extensions will eventually import additional external dependencies and/or internal libraries, and the only way to prevent our settings/options from leaking against unrelated targets is to always use target-based properties

Reply:
Thanks, @alessandrogario for the details. I moved it to `target_compile_definitions` and defined an interface library. The option flag will get added on the usage of the extension interface functions.